src/main/java/com/vestmark/bitbucket/plugin/MergeConflictDetectorServlet.java

```diff
@@ -143,31 +172,149 @@ public void dryRunMerge(MergeConflictDetector mcd, Branch toBranch)
       if (result != null) {
         message.add("Merge committed! Commit ID: " + result.getLatestCommit());
       }
-    } catch (Exception e) {
+      doAheadCheck(toBranch.getId(), mcd.getToRepo(), defBranchId);
+    }
+    catch (Exception e) {
       if (e instanceof MergeException) {
         files = new LinkedList<String>();
         mergeConflicts = new LinkedList<GitMergeConflict>();
-        // When a merge cannot be completed automatically, a CommandFailedException is being thrown and caught in this MergeException block by mistake.
+        // When a merge cannot be completed automatically, a CommandFailedException is being thrown and caught in this
+        // MergeException block by mistake.
         // The (GitMergeException) cast below causes the plugin to crash because of it.
         // Adding a CommandFailedException catch block did not work.
         // Checking the type of the Exception obj prior to the cast using instanceof did not work.
         // Encasing the cast inside its own try/catch was the only way I could find to keep the plugin from crashing.
         try {
-          for (GitMergeConflict mergeConflict : ((GitMergeException)e.getCause()).getConflicts()) {
+          for (GitMergeConflict mergeConflict : ((GitMergeException) e.getCause()).getConflicts()) {
             files.add(mergeConflict.getMessage().replaceFirst("Merge conflict in ", ""));
-            message.add("Source change: " + mergeConflict.getOurChange() + " Target change: " 
-                                          + mergeConflict.getTheirChange());
+            message.add(
+                "Source change: " + mergeConflict.getOurChange() + " Target change: " + mergeConflict.getTheirChange());
             mergeConflicts.add(mergeConflict);
           }
-        } catch (Exception f) {
+        }
+        catch (Exception f) {
           message.add(e.getMessage());
         }
-      } else {
-        // Non Merge Exception
-        message.add(e.getMessage());
+      }
+      else {
+        message.add("Non-Merge Exception: " + e.getMessage());
       }
     }
     mcd.addResult(toBranch, mergeConflicts, message, files);
   }
-}
 
+  private void doAheadCheck(String fromBranchId, Repository repository, String toId) throws MergeException
```

Comment:
this is always attempting to merge to master?
why not check for automatic merge failure PR's instead?

Reply:
This is simply calling to get a diff on the release branch to master branch. It is only called if no merge conflicts are found.
If any release branch is found to be ahead of the master branch, this is what we're currently looking at to indicate that there may be an auto merge failure PR out there. 
The other way would be to list out all open PRs and find one that is created for an automerge failure, but that may be a bit more intensive..

Reply:
enumerating open pull requests should be less resource intensive and provide the ability to accurately report - instead of a maybe.

Reply:
well.. maybe it's just language :), but seeing that a release branch is ahead of master is a very sure way that there is something wrong with the auto-merges. I'm trying the listing out pull requests to see for sure which is more intensive.